File: Array/C++/Array.cpp
#include<iostream>
#include<string>

using namespace std;


int main(){
    const int size = 4;

    int intgerArr[size] = {1, 2, 3, 4};

    for (int i = 0; i < size; i++)
    {
        cout << intgerArr[i];
    }

    cout << "\n";

    //other types of Array
    // String array
    const char *strings[size] = {"vin", "pat", "las", "masa"};

    for (int i = 0; i < size; i++){

        cout << strings[i] << "\t";
    }

    cout << "\n";

    // Other String

    string name[size] = {"leya", "jaya", "miya"};

    for (int i = 0; i < size; i++)
    {
        cout << name[i] << "\t";
    }

    cout << "\n";

    // Other arrays 
    // float

    float floating[size] = {2.3, 4.3, 5.5}; 

    for (int i = 0; i < size; i++)
    {
        cout << floating[i] << "\t";
    }
    
    cout << "\n";

    // Double 

    double doubleArr[size] = {4.5, 63.4, 4.55, 0.0};

    for (int i = 0; i < size; i++)
    {
        cout << doubleArr[i] << "\t";
    }

    cout << "\n";

    // Boolean

    bool BooleanArr[size] = {true, false, true};

    for (int i = 0; i < size; i++)
    {
        cout << BooleanArr[i] << "\t";
    }
    
    return 0;
}

File: 1. Array/Kotlin/array.kt
fun main(){

    //Arrays in kotlin

    // Integer Array

    val integerArr:IntArray = intArrayOf(1, 2, 3, 3)

    // println(integerArr.size) // size/length of Array

    for(i in integerArr){
        println(i)
    }

    // floating Array

    val floatArr:FloatArray = floatArrayOf(2.3f, 3.3f, 6.7f, 5.9f)
    for (i in floatArr) {
        println(i)
    }

    // Double Array

    val doubleArr:DoubleArray = doubleArrayOf(2.444444, 5.65, 6.000)
    for (i in doubleArr) {
        println(i)
    }

    // String Array

    val stringArr:Array<String> = arrayOf("jaya", "leya", "maya")
    for (i in stringArr) {
        println(i)
    }

    // Other types of Array

    val charArr:CharArray = charArrayOf('d', 'm', 'f', 'k')
    for(i in charArr){
        println(i)
    }

    // ByteArray

    val byteArr:ByteArray = byteArrayOf(0x2E, 0x38)
    for(i in byteArr){
        println(i)
    }


}